### ISSUE TITLE
Refuse to register if there is no compat entry for Julia

### ISSUE THREAD

**Issue by DilumAluthge** [JuliaRegistries/Registrator.jl#258]
We have a long list of recommendations/guidelines for automerging. If your package doesn't meet those recommendations/guidelines, it can still be manually merged.

But the rule that all packages must have a `[compat]` entry for Julia isn't really one of the automerge guidelines. It's a hard rule that we enforce for all packages. I.e. I won't manually merge a package if it doesn't have a `[compat]` entry for Julia.

So in that case, I actually think that Registrator should enforce this by refusing to register a package if it doesn't have a `[compat]` entry for Julia.

**Comment by StefanKarpinski** [comment]
Yes, I think of `julia` as an implicit dependency of all projects, so we should require a compat entry for it in order to automerge, like we do for other dependencies.

**Comment by DilumAluthge** [comment]
> Yes, I think of `julia` as an implicit dependency of all projects, so we should require a compat entry for it in order to automerge, like we do for other dependencies.

I should clarify. It is already the case that a Julia compat entry is required for automerge.

What I am saying is that we should require a Julia compat entry even for manual merge.

**Comment by StefanKarpinski** [comment]
Oh, sure. That seems fine as well.
